### ISSUE TITLE
[Bug]: Rounding in intermediate math introduces error

### ISSUE THREAD

**Issue by Sean-Hastings** [textstat/textstat#202]
### Operating system

Linux

### Download source

GitHub clone

### Python distribution

CPython

### Python version

Python 3.12.4

### What happened?

Because many metrics (eg. `textstat.flesch_reading_ease`) use other `textstat` metrics as intermediate math, the rounding in those other metric functions introduces error into the "compound" metrics.

Example provided in the test section below. It gets 1.447 syllables/word for the base text and 1.454 with a single 4-syllable word added. With the default rounding, this gets exaggerated to 1.4 vs 1.5. The FRE then goes from 63.10 and 62.47 un-rounded to 67.1 and 58.5 with rounding enabled. That's a difference of ~15% where the difference should be negligible.

### Other information

None

### Failing tests

```python
long_text = (
    "Playing ... games has always been thought to be "
    "important to the development of well-balanced and "
    "creative children; however, what part, if any, "
    "they should play in the lives of adults has never "
    "been researched that deeply. I believe that "
    "playing games is every bit as important for adults "
    "as for children. Not only is taking time out to "
    "play games with our children and other adults "
    "valuable to building interpersonal relationships "
    "but is also a wonderful way to release built up "
    "tension.\n"
    "There's nothing my husband enjoys more after a "
    "hard day of work than to come home and play a game "
    "of Chess with someone. This enables him to unwind "
    "from the day's activities and to discuss the highs "
    "and lows of the day in a non-threatening, kick back "
    "environment. One of my most memorable wedding "
    "gifts, a Backgammon set, was received by a close "
    "friend. I asked him why in the world he had given "
    "us such a gift. He replied that he felt that an "
    "important aspect of marriage was for a couple to "
    "never quit playing games together. Over the years, "
    "as I have come to purchase and play, with other "
    "couples & coworkers, many games like: Monopoly, "
    "Chutes & Ladders, Mastermind, Dweebs, Geeks, & "
    "Weirdos, etc. I can reflect on the integral part "
    "they have played in our weekends and our "
    "\"shut-off the T.V. and do something more "
    "stimulating\" weeks. They have enriched my life and "
    "made it more interesting. Sadly, many adults "
    "forget that games even exist and have put them "
    "away in the cupboards, forgotten until the "
    "grandchildren come over.\n"
    "All too often, adults get so caught up in working "
    "to pay the bills and keeping up with the "
    "\"Joneses'\" that they neglect to harness the fun "
    "in life; the fun that can be the reward of "
    "enjoying a relaxing game with another person. It "
    "has been said that \"man is that he might have "
    "joy\" but all too often we skate through life "
    "without much of it. Playing games allows us to: "
    "relax, learn something new and stimulating, "
    "interact with people on a different more "
    "comfortable level, and to enjoy non-threatening "
    "competition. For these reasons, adults should "
    "place a higher priority on playing games in their "
    "lives. competition readability suitability sinkability"
    "arithmetical development interpersonal"
)
longer_text = long_text + " relationships"
from textstat import textstat as ts

ts.set_rounding(False, None)
ts._cache_clear()

print(f"un-rounded syllable rate on base text: {ts.avg_syllables_per_word(long_text)}")

ts.set_rounding(True, 1)
ts._cache_clear()

print(f"rounded syllable rate on base text: {ts.avg_syllables_per_word(long_text)}")
ts.set_rounding(False, None)
ts._cache_clear()

print(f"un-rounded syllable rate with one extra word: {ts.avg_syllables_per_word(longer_text)}")

ts.set_rounding(True, 1)
ts._cache_clear()

print(f"rounded syllable rate with one extra word: {ts.avg_syllables_per_word(longer_text)}")
ts.set_rounding(False, None)
ts._cache_clear()

print(f"un-rounded FRE on base text: {ts.flesch_reading_ease(long_text)}")

ts.set_rounding(True, 1)
ts._cache_clear()

print(f"rounded FRE on base text: {ts.flesch_reading_ease(long_text)}")
ts.set_rounding(False, None)
ts._cache_clear()

print(f"un-rounded FRE with one extra word: {ts.flesch_reading_ease(longer_text)}")

ts.set_rounding(True, 1)
ts._cache_clear()

print(f"rounded FRE with one extra word: {ts.flesch_reading_ease(longer_text)}")
```

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/textstat/community/blob/main/CODE_OF_CONDUCT.md)

**Comment by alxwrd** [comment]
@Sean-Hastings good catch, thank you for reporting!

I've seen your PR - I'll #201 get merged before I review.

**Comment by Sean-Hastings** [comment]
Sounds good thank you, I'll make sure to update the new PR after #201 gets merged as well.
